src/Controller/LazyControllerFactory.php

```diff
@@ -0,0 +1,154 @@
+     * required for those services with no entry based on the class/interface
+     * name.
+     *
+     * @var string[]
+     */
+    private $aliases = [
+        'Zend\Console\Adapter\AdapterInterface'     => 'ConsoleAdapter',
+        'Zend\Filter\FilterPluginManager'           => 'FilterManager',
+        'Zend\Hydrator\HydratorPluginManager'       => 'HydratorManager',
+        'Zend\InputFilter\InputFilterPluginManager' => 'InputFilterManager',
+        'Zend\Log\FilterPluginManager'              => 'LogFilterManager',
+        'Zend\Log\FormatterPluginManager'           => 'LogFormatterManager',
+        'Zend\Log\ProcessorPluginManager'           => 'LogProcessorManager',
+        'Zend\Log\WriterPluginManager'              => 'LogWriterManager',
+        'Zend\Serializer\AdapterPluginManager'      => 'SerializerAdapterManager',
+        'Zend\Validator\ValidatorPluginManager'     => 'ValidatorManager',
+    ];
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return DispatchableInterface
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $reflectionClass = new ReflectionClass($requestedName);
+
+        if (null === ($constructor = $reflectionClass->getConstructor())) {
+            return new $requestedName();
+        }
+
+        $reflectionParameters = $constructor->getParameters();
+
+        if (empty($reflectionParameters)) {
+            return new $requestedName();
+        }
+
+        $parameters = [];
+        foreach ($reflectionParameters as $parameter) {
+            if ($parameter->isArray()
+                && $parameter->getName() === 'config'
+                && $container->has('config')
+            ) {
+                $parameters[] = $container->get('config');
+                continue;
+            }
+
+            if ($parameter->isArray()) {
+                $parameters[] = [];
+                continue;
+            }
+
+            if (! $parameter->getClass()) {
+                $parameters[] = null;
+                continue;
+            }
+
+            $type = $parameter->getClass()->getName();
+            $type = array_key_exists($type, $this->aliases) ? $this->aliases[$type] : $type;
+
+            if (! $container->has($type)) {
+                throw new ServiceNotFoundException(sprintf(
+                    'Unable to create controller "%s"; unable to resolve parameter "%s" using type hint "%s"',
+                    $requestedName,
+                    $parameter->getName(),
+                    $type
+                ));
+            }
+
+            $parameters[] = $container->get($type);
+        }
+
+        return $reflectionClass->newInstanceArgs($parameters);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function canCreate(ContainerInterface $container, $requestedName)
+    {
+        if (! is_string($requestedName) || ! class_exists($requestedName)) {
```

Comment:
I've seen `null` a few times, though primarily when used in a servicemanager v2 context. I'll remove that check.